### PR TITLE
chore: update h1 titles and meta descriptions

### PR DESCRIPTION
**Pull request by nensidosari**
## Changes
- update h1 titles and meta descriptions

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

MI-590 #done

### Preview domain
https://mi-590-seo.preview.app.daily.dev

### CODE REVIEW

File: packages/webapp/pages/upvoted.tsx [hunk 1]
```diff
@@ -5,12 +5,14 @@ import {
   getMainFeedLayout,
   mainFeedLayoutProps,
 } from '../components/layouts/MainFeedPage';
-import { defaultOpenGraph, defaultSeo } from '../next-seo';
+import { defaultOpenGraph } from '../next-seo';
+import { getTemplatedTitle } from '../components/layouts/utils';
 
 const seo: NextSeoProps = {
-  title: 'Most upvoted posts on daily.dev',
+  title: getTemplatedTitle('Most upvoted posts for developers'),
```
Comment (nensidosari) [inline packages/webapp/pages/upvoted.tsx:12]: This change is not in the specs doc, so waiting to confirm here: https://dailydotdev.slack.com/archives/C06JCC41KSS/p1728371071348349?thread_ts=1728212815.310049&cid=C06JCC41KSS

File: packages/webapp/components/layouts/ProfileLayout/index.tsx [hunk 2]
```diff
@@ -62,8 +62,27 @@ export default function ProfileLayout({
   ogImageUrl.searchParams.set('type', 'wide');
   ogImageUrl.searchParams.set('r', Math.random().toString(36).substring(2, 5));
 
+  const getSeoTitle = () => {
```
Comment (rebelchris) [inline packages/webapp/components/layouts/ProfileLayout/index.tsx:65]: We could also move this to the individual pages rather maybe?
then you don't need the switch statements here?

File: packages/webapp/pages/squads/discover/featured.tsx [hunk 3]
```diff
@@ -27,10 +27,9 @@ export type Props = {
 };
 
 const seo: NextSeoProps = {
-  title: getTemplatedTitle('Featured Squads'),
+  title: getTemplatedTitle('Explore the featured Squads'),
   openGraph: { ...defaultOpenGraph },
-  ...defaultSeo,
-  description: `Dive into daily.dev's Featured Squads, showcasing editor's choice and hand-picked communities that highlight the most innovative and engaging groups on the platform. Join the conversation with top developers today.`,
+  description: `Explore daily.dev’s featured squads, handpicked by our editors. Join the best developer communities and engage in top discussions today.`,
```
Comment (rebelchris) [inline packages/webapp/pages/squads/discover/featured.tsx:32]:
```suggestion
  description: `Explore daily.dev’s featured Squads, handpicked by our editors. Join the best developer communities and engage in top discussions today.`,
```

File: packages/webapp/pages/squads/discover/index.tsx [hunk 4]
```diff
@@ -9,16 +9,25 @@ import {
...
 interface SquadDirectoryTitleProps {
   icon?: boolean;
   title: string;
 }
 
+const seo: NextSeoProps = {
+  title: getTemplatedTitle('Explore all Squads'),
+  openGraph: { ...defaultOpenGraph },
+  description:
+    'Browse and join squads on daily.dev. Connect with fellow developers, share knowledge, and dive into specific topics of interest in your favorite squads.',
```
Comment (rebelchris) [inline packages/webapp/pages/squads/discover/index.tsx:28]:
```suggestion
    'Browse and join Squads on daily.dev. Connect with fellow developers, share knowledge, and dive into specific topics of interest in your favorite Squads.',
```

File: packages/webapp/pages/squads/discover/[id].tsx [hunk 5]
```diff
@@ -51,12 +51,14 @@ function SquadCategoryPage({ category }: SquadCategoryPageProps): ReactElement {
     result.data?.pages.flatMap((page) => page.sources.edges) ?? [];
   const isTablet = useViewSize(ViewSize.Tablet);
 
-  const title = `${category?.title} Directory`;
-  const description = `Explore the ${category?.title} Squads on daily.dev, where developers share insights, collaborate on projects, and discuss the latest trends. Join a squad that matches your interests and elevate your developer journey.`;
+  const seo = {
+    title: getTemplatedTitle(`Explore ${category?.title} Squads`),
+    description: `Find the best squads in the ${category?.title} category on daily.dev. Connect with like-minded developers and collaborate on the latest technologies.`,
```
Comment (rebelchris) [inline packages/webapp/pages/squads/discover/[id].tsx:56]:
```suggestion
    description: `Find the best Squads in the ${category?.title} category on daily.dev. Connect with like-minded developers and collaborate on the latest technologies.`,
```

**Review by rebelchris — CHANGES_REQUESTED** [review]
Looks good, minor feedback on `Squads` usage

**Review by rebelchris — APPROVED** [review]

**Review by omBratteng — APPROVED** [review]

**Pull request merged**